test_conformance/c11_atomics/test_atomics.cpp

```diff
@@ -2146,6 +2146,218 @@ class CBasicTestFetchMin
+                           * (ref_vals.data() < spec_vals.data()
+                                  ? threadCount
+                                  : spec_vals.size()));
+            }
+
+            memcpy(startRefValues, ref_vals.data(),
+                   sizeof(HostDataType) * threadCount);
+
+            return true;
+        }
+        return false;
+    }
+    std::string ProgramCore() override
+    {
+        std::string memoryOrderScope = MemoryOrderScopeStr();
+        std::string postfix(memoryOrderScope.empty() ? "" : "_explicit");
+        return "  oldValues[tid] = atomic_fetch_min" + postfix
+            + "(&destMemory[tid], oldValues[tid] " + memoryOrderScope + ");\n";
+    }
+    void HostFunction(cl_uint tid, cl_uint threadCount,
+                      volatile HostAtomicType *destMemory,
+                      HostDataType *oldValues) override
+    {
+        oldValues[tid] = host_atomic_fetch_min(&destMemory[tid], oldValues[tid],
+                                               MemoryOrder());
+    }
+    bool ExpectedValue(HostDataType &expected, cl_uint threadCount,
+                       HostDataType *startRefValues,
+                       cl_uint whichDestValue) override
+    {
+        expected = StartValue();
+        if constexpr (std::is_same<HostDataType, HOST_ATOMIC_DOUBLE>::value)
+        {
+            if (startRefValues[whichDestValue] < expected)
+                expected = startRefValues[whichDestValue];
+        }
+        return true;
+    }
+
+    bool VerifyExpected(const HostDataType &expected,
+                        const HostAtomicType *const testValue,
+                        const HostDataType *startRefValues,
+                        cl_uint whichDestValue) override
+    {
+        if (testValue != nullptr && startRefValues != nullptr)
+        {
+            if (testValue[whichDestValue] != expected)
+            {
+                // special cases
+                // min(-0, +0) = min(+0, -0) = +0 or -0,
+                if (((startRefValues[whichDestValue] == -0.f)
+                     && (StartValue() == 0.f))
+                    || ((startRefValues[whichDestValue] == 0.f)
+                        && (StartValue() == -0.f)))
+                    return false;
+                else if (is_qnan(StartValue())
+                         || is_qnan(startRefValues[whichDestValue]))
+                {
+                    // min(x, qNaN) = min(qNaN, x) = x,
+                    // min(qNaN, qNaN) = qNaN,
+                    if (is_qnan(StartValue())
+                        && is_qnan(startRefValues[whichDestValue]))
+                        return !is_qnan(testValue[whichDestValue]);
+                    else if (is_qnan(StartValue()))
+                        return testValue[whichDestValue]
+                            != startRefValues[whichDestValue];
+                    else
+                        return testValue[whichDestValue] != StartValue();
```

Comment:
@bashbaug the same problem as [here](https://github.com/KhronosGroup/OpenCL-CTS/pull/2391#discussion_r2090623307)